internal/validators/oci_validator.go

```diff
@@ -76,7 +77,15 @@ func (s *OciRuleService) ReconcileOciRegistryRule(rule v1alpha1.OciRegistryRule,
 
 	errMsg := "failed to validate artifact in registry"
 	for _, artifact := range rule.Artifacts {
-		detail, err := validateArtifact(ctx, rule.Host, artifact, opts, vr)
+		ref, err := generateRef(rule.Host, artifact.Ref, vr)
+		if err != nil {
+			details = append(details, fmt.Sprintf("failed to generate reference for artifact %v/%v", rule.Host, artifact.Ref))
```

Comment:
Now I'm noticing that you have `%v`s everywhere for formatting strings. I don't expect you to change them all... but I am not sure that that is idiomatic.

Reply:
Updated all instances of %v that I've introduced mostly to match patterns used in our repos.